h2o-parsers/h2o-parquet-parser/src/main/java/water/parser/parquet/ChunkConverter.java

```diff
@@ -84,22 +84,35 @@ private PrimitiveConverter newConverter(int colIdx, byte vecType, PrimitiveType
   }
 
   private static class StringConverter extends PrimitiveConverter {
-
+    // Maximum array size (Integer.MAX_VALUE - 8) minus one place for a trailing zero
+    private static final int MAX_STRING_LENGTH = Integer.MAX_VALUE - 9;
     private final BufferedString _bs = new BufferedString();
     private final int _colIdx;
     private final WriterDelegate _writer;
     private final boolean _dictionarySupport;
     private String[] _dict;
+    private int _writtenAmount;
+    private boolean _overflew;
 
     StringConverter(WriterDelegate writer, int colIdx, boolean dictionarySupport) {
       _colIdx = colIdx;
       _writer = writer;
       _dictionarySupport = dictionarySupport;
+      _writtenAmount = 0;
+      _overflew = false;
     }
 
     @Override
     public void addBinary(Binary value) {
+      if(_overflew) return;
       _bs.set(StringUtils.bytesOf(value.toStringUsingUTF8()));
+      _writtenAmount += _bs.length();
+      int difference = _writtenAmount - MAX_STRING_LENGTH;
+      if(difference > 0){
+        _bs.setLen(_bs.length() - difference);
+        _overflew = true;
+        return;
+      }
```

Comment:
We should report back to the user that we didn't parse the whole thing. At least logging the error would be a good idea

Reply:
Added Log.info("A String chunk value overflext maximum allowed size by ", lenDifference, " bytes. Stripping the string.\n");